File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    """
    Keep track of the mx value of the difference between the maximum value and the current stock prices. Now if the difference between the mx value is > profit then we can update the profit
    return profit.
    
    mn = minimum stock price encountered
    profit = maximum profit that we can make.

    """
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0;mn = sys.maxsize;
        for i in prices:
            if i<mn :
                mn = i
            newProfit = i - mn;
            if newProfit > profit:
                profit = newProfit;
        return profit;

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
/*
    Keep track of the mx value of the difference between the maximum value and the current stock prices. Now if the difference between the mx value is > profit then we can update the profit
    return profit.
    
    mn = minimum stock price encountered
    profit = maximum profit that we can make.

*/
    int maxProfit(vector<int>& prices) {
        int mn = INT_MAX,profit = 0;
        for (int i = 0;i<prices.size();i++){
            if (prices[i]<mn) mn = prices[i];
            if (prices[i]-mn>profit) profit = prices[i]-mn;
        }
        return profit;
    }
};